### PR TITLE
fix(lib/babe): add pre-runtime digest before calling initialize_block

### PR DESCRIPTION
**Pull request by noot**
## Changes

- fix creation of `NextEpochData` digest by adding pre-runtime digest to block before calling `initialize_block`

## Tests

```
go test ./lib/babe
```

## Issues

- related to #1479

### CODE REVIEW

**Review by edwardmack — APPROVED** [review]

File: lib/babe/build.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func (b *Service) buildBlock(parent *types.Header, slot Slot) (*types.Block, err
 
 	// create new block header
 	number := big.NewInt(0).Add(parent.Number, big.NewInt(1))
-	header, err := types.NewHeader(parent.Hash(), common.Hash{}, common.Hash{}, number, types.NewEmptyDigest())
+	header, err := types.NewHeader(parent.Hash(), common.Hash{}, common.Hash{}, number, types.Digest{preDigest})
```
Comment (timwu20) [inline lib/babe/build.go:52]: nit: create a new constructor for this?

**Review by timwu20 — APPROVED** [review]

File: lib/babe/build.go (hunk 1 above)
Comment (noot) [reply lib/babe/build.go:52]: done!

File: dot/types/digest.go [hunk 2]
```diff
@@ -34,6 +34,11 @@ func NewEmptyDigest() Digest {
 	return []DigestItem{}
 }
 
+// NewDigest returns a new Digest from the given DigestItems
+func NewDigest(items ...DigestItem) Digest {
+	return Digest(items)
```
Comment (arijitAD) [inline dot/types/digest.go:39]: You can directly do
```
return items
```

**Review by arijitAD — APPROVED** [review]

File: dot/types/digest.go (hunk 2 above)
Comment (arijitAD) [inline dot/types/digest.go:39]:
```suggestion
	return items
```

**Review by arijitAD — APPROVED** [review]

**Pull request merged**